### PR TITLE
Rename SearchableSnapshotIndexInput and move cache classes into store package

### PR DESCRIPTION
**Pull request by tlrx**
This pull request suggests to rename the `SearchableSnapshotIndexInput` class to `DirectBufferedIndexInput` (in comparison to `CacheBufferedIndexInput`) and places it into a `org.elasticsearch.index.store.direct` package.

It also moves `CacheBufferedIndexInput` and associated classes to the `org.elasticsearch.index.store.cache` package so that it is colocated with the `direct` package.

The `IndexInputStats` class is moved to `org.elasticsearch.index.store`.

### CODE REVIEW

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by tlrx** [comment]
@elasticmachine update branch

**Review by DaveCTurner — COMMENTED** [review]
My concern with `DirectBufferedIndexInput` is that it sounds very general and, with this change, is not even in a package suggesting its relationship to searchable snapshots. How about `DirectBlobContainerIndexInput` vs `CachedBlobContainerIndexInput`?

**Comment by tlrx** [comment]
> How about `DirectBlobContainerIndexInput` vs `CachedBlobContainerIndexInput`?

This is a good suggestion, I use it to rename the classes. Let me know what you think.

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/SparseFileTrackerTests.java [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
  * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.xpack.searchablesnapshots.cache;
+package org.elasticsearch.index.store.cache;
 
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.action.ActionListener;
 import org.elasticsearch.cluster.coordination.DeterministicTaskQueue;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.store.cache.SparseFileTracker;
```
Comment (DaveCTurner) [inline x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/SparseFileTrackerTests.java]: Strange, this import is unnecessary but checkstyle is happy.

**Review by DaveCTurner — APPROVED** [review]
LGTM, but for one nit.

File: x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/SparseFileTrackerTests.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/cache/SparseFileTrackerTests.java]: Looks like a checkstyle limitation :/ I removed it in df9bc92

**Comment by tlrx** [comment]
Thanks David

**Pull request merged**